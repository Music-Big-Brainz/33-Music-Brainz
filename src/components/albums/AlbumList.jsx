import React from 'react';
import PropTypes from 'prop-types';
import Album from './Album';

<<<<<<< HEAD
const AlbumList = ({ releases }) => {
=======

const AlbumList = ({ releases, name }) => {
>>>>>>> 9f3bac0bb35917b144bc4e6f3c2f061a4f6f9ba5
  const albumElements = releases.map(release => (

    <li key={release.id}>
      <Album {...release} />

    <li key={release.albumId}>
      <Album {...release} name={name} />

    </li>
  ));

  return <ul data-testid="albums">{albumElements}</ul>;
};

AlbumList.propTypes = {
  name: PropTypes.string.isRequired,
  releases: PropTypes.arrayOf(
    PropTypes.shape({
      albumId: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      date: PropTypes.string.isRequired,
    })
  )
};

export default AlbumList;
