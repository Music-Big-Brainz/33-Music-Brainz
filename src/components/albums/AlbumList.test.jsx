import React from 'react';
import { render, cleanup } from '@testing-library/react';
import AlbumList from './AlbumList';
import MemoryRouter from 'react-router-dom';

describe('displays a list of albums', () => {
  afterEach(() => cleanup());
  it('renders a list of albums', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <AlbumList title="" date="" />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
