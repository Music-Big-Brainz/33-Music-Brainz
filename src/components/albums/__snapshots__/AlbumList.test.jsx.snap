// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a list of albums renders a list of albums 1`] = `
<DocumentFragment>
  <ul
    data-testid="albums"
  />
</DocumentFragment>
`;
